tests/ChannelTest.php

```diff
@@ -13,14 +15,172 @@ class ChannelTest extends TestCase
     /**
      * Test channel link leads to route
      */
-    public function testChannelLink()
+    public function testUserCanViewChannels()
     {
-        $this->createUser(1);
-        $this->createChannel();
-        $this->createEpisode();
+        $channel = factory(Channel::class)->create();
+        $this->visit(route('channels'))
+             ->see($channel['channel_name']);
 
-        $this->visit('/channels')
-             ->seePageIs('/channels');
+        $this->call('GET', '/channels');
+        $this->assertViewHasAll(['channels']);
+
+    }
+
+    /**
+     * Assert that a new channel is created
+     * in the database.
+     *
+     * @return void
+     */
+    public function testCreateChannel()
+    {
+        $this->withoutMiddleware();
+
+        $user = factory(User::class)->create(['role_id' => 3]);
+        $this->actingAs($user)
+         ->visit('/dashboard/channel/create')
+         ->type('Swanky new name', 'name')
+         ->type('Swanky new description', 'description')
+         ->press('create')
+         ->seeInDatabase('channels', [
+            'channel_name' => 'Swanky new name'
+         ]);
+
+         $user = factory(User::class)->create();
```

Comment:
**Code Functionality - Critical**

Only  a user with the role of an admin or premium user should have the ability to create channels